### DISCUSSION TITLE
Codespaces update has broken Golang

### DISCUSSION THREAD

**Discussion by pdeva** [community/community#46707] [Codespaces]
### Select Topic Area

Bug

### Body

to repro, create codespace using `mcr.microsoft.com/devcontainers/universal:2-focal` docker image.

it now puts `$GOPATH` at `/go` which has permissions like this:

```
drwxrwsr-x 1 codespace golang 4.0K Feb  2 23:47 bin
drwxr-sr-x 4 root      golang 4.0K Feb  2 23:44 pkg
```

thus running running `go get -u ./...` is resulting in permission denied errors.

```
go: writing go.mod cache: mkdir /go/pkg/mod/cache/download/github.com/!data!dog: permission denied
go: writing go.mod cache: mkdir /go/pkg/mod/cache/download/github.com/aws: permission denied
go: writing go.mod cache: mkdir /go/pkg/mod/cache/download/github.com/aws: permission denied
```

this seems to have been a recent change since everything was working fine a week or so ago.

**Comment by Mannyflow7** [comment]
1. [](`[README.md](https://github.com/community/community/files/10682890/README.md)`)

- > 

- [ ]  @[](url)
[pull_request_template.md](https://github.com/community/community/files/10682977/pull_request_template.md)
 ##
![qrcode_apireference getresponse com](https://user-images.githubusercontent.com/119740633/217463442-de1fe4c6-25f5-4401-8c71-98525f1d9b68.png)https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax
@ @ @ @
### 
[README.md](https://github.com/community/community/files/10682956/README.md)
**
[README (2).md](https://github.com/community/community/files/10682961/README.2.md)

> **

- [ ] 

- [ ] 
![qrcode_apireference getresponse com](https://user-images.githubusercontent.com/119740633/217463590-e1f3dbe3-f0fd-4281-8117-d8026bcd5f37.png)

**Reply by Mannyflow7** [reply]
[pull_request_template.md](https://github.com/community/community/files/10682995/pull_request_template.md)
[pull_request_template.md](https://github.com/community/community/files/10682996/pull_request_template.md)
### 
![qrcode_apireference getresponse com](https://user-images.githubusercontent.com/119740633/217465312-0b98c432-053f-42b5-8614-e77340d9372e.png)
**
[magento-20.5.7 (5).tar.gz](https://github.com/community/community/files/10683020/magento-20.5.7.5.tar.gz)
_
[GPO-CONAN-2020-SUPP-1.pdf](https://github.com/community/community/files/10683049/GPO-CONAN-2020-SUPP-1.pdf)
[**_](![unnamed](https://user-images.githubusercontent.com/119740633/217465871-895e5678-d882-4566-a8fd-283b9ecbb7bc.png))

**Comment by joshaber** [comment]
Hi @pdeva, I just tried to reproduce this and was unable. What I did:
1. Create a codespace from https://github.com/joshaber/simple-server/tree/golang-test
2. Run `go install github.com/haccer/subjack@latest`
3. Verify it works
4. Verify permissions under `$GOPATH`
```
$ ls -la $GOPATH
total 20
drwxrwsr-x 1 codespace golang 4096 Feb  8 13:50 .
drwxr-xr-x 1 root      root   4096 Feb  8 13:47 ..
drwxrwsr-x 1 codespace golang 4096 Feb  8 13:51 bin
drwxr-sr-x 4 codespace golang 4096 Feb  8 13:50 pkg
```

Is it possible something is running `go get ...` root instead of the current user?

**Reply by pdeva** [reply]
what is the value of `$GOPATH` for you?

**Reply by joshaber** [reply]
`/go`

**Comment by pdeva** [comment]
[devcontainer.zip](https://github.com/community/community/files/10688410/devcontainer.zip)
attaching our devcontainer dir to allow you to reproduce the issue

**Comment by joshaber** [comment]
I'll defer to the experts over at https://github.com/devcontainers/images/issues/405

**Comment by rayjlinden** [comment]
Having this same problem....
